DSharpPlus/BaseExtension.cs

```diff
@@ -21,18 +21,22 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
+using System;
+
 namespace DSharpPlus
 {
     /// <summary>
     /// Represents base for all DSharpPlus extensions. To implement your own extension, extend this class, and implement its abstract members.
     /// </summary>
-    public abstract class BaseExtension
+    public abstract class BaseExtension : IDisposable
     {
         /// <summary>
         /// Gets the instance of <see cref="DiscordClient"/> this extension is attached to.
         /// </summary>
         public DiscordClient Client { get; protected set; }
 
+        public abstract void Dispose();
```

Comment:
as-is, user extensions will have to implement Dispose themselves. Consider changing this to `public virtual void Dispose() {}`

Reply:
well, it violates the last line of #1509 - it is indeed a breaking change, albeit for a very small amount of people

Reply:
hmm. We could announce the change? v4.4 has plenty of breaking changes by now